service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java

```diff
@@ -850,11 +852,17 @@ public MicroserviceInstance findServiceInstance(String serviceId, String instanc
           new RequestParam().addHeader("X-ConsumerId", serviceId).addQueryParam("global", "true"),
           syncHandler(countDownLatch, MicroserviceInstanceResponse.class, holder));
       countDownLatch.await();
+      if (holder.hasException()) {
+        throw holder.getThrowable();
+      }
       if (null != holder.value) {
         return holder.value.getInstance();
       }
       return null;
-    } catch (Exception e) {
+    } catch (ConnectException e1) {
+      LOGGER.error("Find service instance from service center failed.", e1);
```

Comment:
Not check all the usage of findServiceInstance, this added logic will cause other parts change.